### PR TITLE
test: add tests for `TransactionNotifier`

### LINKED ISSUE #291: Create tests for `TransactionNotifier`
- using the existing test suite as an example, create tests to stress-test `TransactionNotifier`
- the `TransactionNotifier` widget is responsible for:
  - polling PFIs for a specific exchange id until an exchange is considered closed or until it is closed by the caller which started the polling

### PR DESCRIPTION
**Pull request by victoreronmosele**
This PR adds UI tests for the `TransactionNotifier` class.

Closes #291.

### CODE REVIEW

File: test/features/transaction/transaction_notifier_test.dart [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:mocktail/mocktail.dart';
+
+import '../../helpers/mocks.dart';
+import '../../helpers/riverpod_helpers.dart';
+import '../../helpers/test_data.dart';
+
+class Listener<T> extends Mock {
+  void call(T? previous, T next);
+}
```
Comment (ethan-tbd) [inline test/features/transaction/transaction_notifier_test.dart:17]: think we can just move this to `mocks.dart`

File: test/features/transaction/transaction_notifier_test.dart [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      const AsyncData<Transaction?>(null),
+    );
+  });
+
+  group('TransactionNotifier', () {
+    final pfi = TestData.getPfi('did:dht:pfiDid');
+    const exchangeId = 'rfq_01ha835rhefwmagsknrrhvaa0k';
+    final parameters = TransactionProviderParameters(pfi, exchangeId);
+
+    final did = TestData.aliceDid;
```
Comment (ethan-tbd) [inline test/features/transaction/transaction_notifier_test.dart:33]: nit: let's move this to right below the `await TestData.initializeDids()`

File: test/features/transaction/transaction_notifier_test.dart [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      final container = createContainer(
+        overrides: [
+          didProvider.overrideWith(
+            (ref) => did,
+          ),
+          tbdexServiceProvider.overrideWith(
+            (ref) => mockTbdexService,
+          ),
+        ],
+      );
```
Comment (ethan-tbd) [inline test/features/transaction/transaction_notifier_test.dart:57]: nice use of the `createContainer()` helper!

**Comment by victoreronmosele** [comment]
Thanks for the feedback! 

All comments have been addressed.

**Review by ethan-tbd — APPROVED** [review]
great work once again :)

**Pull request merged**